server/index.js

```diff
@@ -85,10 +85,38 @@ app.get("/api/eggs", (req, res, next) => {
   return db
     .query(sql)
     .then((result) => {
-      const [egg] = result.rows;
-      egg.longitude = Number(egg.longitude);
-      egg.latitude = Number(egg.latitude);
-      res.status(200).json([egg]);
+      const egg = result.rows;
+      egg.forEach((egg) => {
+        egg.latitude = Number(egg.latitude);
+        egg.longitude = Number(egg.longitude);
+      });
+      res.status(200).json(egg);
+    })
+    .catch((err) => next(err));
+});
+
+app.get("/api/details", (req, res, next) => {
+  const eggId = Number(req.headers.eggid);
```

Comment:
Usually we put information like this in the url path instead of headers `/api/eggs/:eggId`